src/components/settings/SettingsSSOTab/hooks.js

```diff
@@ -75,14 +100,26 @@ const useIdpState = () => {
       // but we need to update this support the case when the correct sso config must be updated
       setProviderConfig(response.data);
 
-      // also get samlproviderdata updated from remote metadata url
-      const providerdataResponse = await updateSamlProviderData({
-        enterpriseId,
-        metadataURL,
-        entityID,
-      });
-      logInfo(providerdataResponse);
+      // If the user already has a provider data entry with a different entityID, remove it before
+      // creating a new one
+      if (existingIdpDataId && (existingIdpDataEntityId !== entityID)) {
+        await deleteSamlProviderData(existingIdpDataId, enterpriseId);
+      }
 
+      if (
+        entryType === 'direct' || (existingMetadataUrl !== metadataURL || existingIdpDataEntityId !== entityID)
+      ) {
+        // also get samlproviderdata updated from remote metadata url
+        const providerdataResponse = await updateSamlProviderData({
+          enterpriseId,
+          metadataURL,
+          entityID,
```

Comment:
NIT: Why is it enterprise**Id** but entity**ID**? I think the first one is probably more proper